cmd/omniwitness/monolith.go

```diff
@@ -54,6 +58,25 @@ func main() {
 	flag.Parse()
 	ctx := context.Background()
 
+	if *metricsAddr == "" {
+		glog.Info("No metrics_listen address provided so skipping prometheus setup")
+		mf := monitoring.InertMetricFactory{}
+		monitoring.SetMetricFactory(mf)
+	} else {
+		mf := prometheus.MetricFactory{
+			Prefix: "omniwitness_",
+		}
+		monitoring.SetMetricFactory(mf)
+
+		go func() {
+			http.Handle("/metrics", promhttp.Handler())
+			if err := http.ListenAndServe(*metricsAddr, nil); err != http.ErrServerClosed {
```

Comment:
I know this isn't going to be "internet facing", but just happened to be reading this post earlier today which suggests that `ListenAndServe` can end up leaking connections & FDs over time due to a lack of timeout being set: https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/#:~:text=http.ListenAndServe%20is%20doing%20it%20wrong

Reply:
Fixed.